include/aws/http/private/h1_encoder.h

```diff
@@ -5,37 +5,19 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
-#include <aws/common/mutex.h>
 #include <aws/http/private/http_impl.h>
 #include <aws/http/private/request_response_impl.h>
 
-struct aws_http1_stream_chunk {
+struct aws_h1_chunk {
+    struct aws_allocator *allocator;
     struct aws_input_stream *data;
-    size_t data_size;
+    uint64_t data_size;
     aws_http1_stream_write_chunk_complete_fn *on_complete;
     void *user_data;
     struct aws_linked_list_node node;
+    /* Buffer containing pre-encoded start line: chunk-size [chunk-ext] CRLF */
     struct aws_byte_buf chunk_line;
-    struct aws_byte_cursor chunk_line_cursor;
-};
-
-struct aws_http1_chunks {
-    struct aws_mutex lock;
-    struct aws_linked_list chunk_list;
-    /* The currently ready for encoding chunk of data. */
-    struct aws_http1_stream_chunk *current_chunk;
-    bool paused;
-};
-
-enum aws_h1_encoder_body_stream_state {
-    AWS_H1_ENCODER_STATE_CHUNK_INIT,
-    AWS_H1_ENCODER_STATE_CHUNK_LINE,
-    AWS_H1_ENCODER_STATE_CHUNK_PAYLOAD,
-    AWS_H1_ENCODER_STATE_CHUNK_END,
-    AWS_H1_ENCODER_STATE_CHUNK_TERMINATED,
```

Comment:
Removing this 2nd state-machine enum and combining into a single state-machine enum